File: 1528-kids-with-the-greatest-number-of-candies/kids-with-the-greatest-number-of-candies.java
class Solution {
    public List<Boolean> kidsWithCandies(int[] candies, int extraCandies) {
        // find kid with the most candies

        // compare each kid + extra to see if its >= kid with max candies
        int maxCandies = 0;

        for(int c : candies){
            maxCandies = Math.max(c, maxCandies);
        }

        ArrayList<Boolean> answer = new ArrayList<>();

        for(int c : candies){
            answer.add(c + extraCandies >= maxCandies);
        }
        return answer;
    }
}

File: 1528-kids-with-the-greatest-number-of-candies/kids-with-the-greatest-number-of-candies.cpp
class Solution {
public:
    vector<bool> kidsWithCandies(vector<int>& candies, int extraCandies) {
        vector<bool> result;
        /*// Find Max Candy
        int max_candy = 0;
        for(int c : candies){
            max_candy = max(c, max_candy);
        }*/

        for(int i = 0; i <= candies.size() - 1; i++){
            if(candies[i] + extraCandies >= *max_element(candies.begin(), candies.end())) result.push_back(true);
            else result.push_back(false);
        }
        return result;
    }
};